wgengine/monitor/monitor_windows.go

```diff
@@ -25,94 +24,67 @@ type eventMessage struct {
 func (eventMessage) ignore() bool { return false }
 
 type winMon struct {
-	logf   logger.Logf
-	ctx    context.Context
-	cancel context.CancelFunc
-
-	mu                    sync.Mutex // guards following
+	logf                  logger.Logf
+	ctx                   context.Context
+	cancel                context.CancelFunc
+	messagec              chan eventMessage
 	addressChangeCallback *winipcfg.UnicastAddressChangeCallback
 	routeChangeCallback   *winipcfg.RouteChangeCallback
-	messagec              chan eventMessage
 }
 
 func newOSMon(logf logger.Logf, _ *Mon) (osMon, error) {
-	m := &winMon{
-		logf: logf,
-	}
+	m := &winMon{logf: logf}
 
-	cba, err := winipcfg.RegisterUnicastAddressChangeCallback(m.unicastAddressChanged)
+	var err error
+	m.addressChangeCallback, err = winipcfg.RegisterUnicastAddressChangeCallback(m.unicastAddressChanged)
 	if err != nil {
 		m.logf("winipcfg.RegisterUnicastAddressChangeCallback error: %v", err)
 		return nil, err
 	}
 
-	cbr, err := winipcfg.RegisterRouteChangeCallback(m.routeChanged)
+	m.routeChangeCallback, err = winipcfg.RegisterRouteChangeCallback(m.routeChanged)
 	if err != nil {
-		cba.Unregister()
+		m.addressChangeCallback.Unregister()
 		m.logf("winipcfg.RegisterRouteChangeCallback error: %v", err)
 		return nil, err
 	}
 
-	m.addressChangeCallback = cba
-	m.routeChangeCallback = cbr
-
 	m.ctx, m.cancel = context.WithCancel(context.Background())
+	m.messagec = make(chan eventMessage, 1)
 
 	return m, nil
 }
 
 func (m *winMon) Close() error {
 	m.cancel()
 
-	m.mu.Lock()
-	defer m.mu.Unlock()
+	var err error
```

Comment:
Bit more conventional is `var ret error` here (ret for return value), and then change err1 to just err below.

then `ret = err` is more obvious in intent (you're setting the return value).

alternatively, just name the return value, since winMon is exported anyway, so it doesn't pollute the godoc:

```go
   func (w *winMon) Close() (ret error) {
```

Reply:
ok

Reply:
Btw. talking about the error returned, what I've meant in my original question was actually should I always return `nil` there, no matter if error while unsubscribing happened or not? Because I don't see anything useful that we can do with this particular error except logging it. We may simply ignore it because the OS will unsubscribe automatically on process exit...